server/src/main/java/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -1417,6 +1408,25 @@ public IPAddressVO associateIPToGuestNetwork(long ipId, long networkId, boolean
         }
     }
 
+    protected boolean isSourceNatAvailableForNetwork(Account owner, IPAddressVO ipToAssoc, Network network) {
```

Comment:
To make it perfect a Java documentation describing when an IP is considered source NAT would be welcome.

Reply:
Well, for me, the java doc should present  the idea of the method, and not just describe the code. The code is the implementation of the idea. By idea here I mean, the context where the given method fits in, the behavior we intend it to have (from an input/output perspective), but not describing the implementation at a line by line perspective.